metrics/exporter.go

```diff
@@ -65,17 +66,31 @@ type ExporterOptions struct {
 	// See https://github.com/knative/serving/blob/master/config/config-observability.yaml
 	// for details.
 	ConfigMap map[string]string
+
+	// A lister for Secrets to allow dynamic configuration of outgoing TLS client cert.
+	Secrets clientv1.SecretLister `json:"-"`
 }
 
 // UpdateExporterFromConfigMap returns a helper func that can be used to update the exporter
 // when a config map is updated.
+// DEPRECATED: Callers should migrate to UpdateExporterFromConfigMap3 (which will be renamed
+// when all callers are migrated).
```

Comment:
Is the rename of UpdateExporterFromConfigMap3 necessary? Why not make it a slightly more descriptive name and then leave it as-is going forward?

Reply:
How about "ConfigMapWatcher"? I've updated this PR.